docs/multimodal/index.md

```diff
@@ -0,0 +1,188 @@
+
+**Pattern Key:**
+
+- **EPD** - All-in-one worker (Simple Aggregated)
+- **E/PD** - Separate encode, combined prefill+decode
+- **E/P/D** - All stages separate
+- **EP/D** - Combined encode+prefill, separate decode
+
+**Status:** ✅ Supported | 🚧 WIP | 🧪 Experimental | ❌ Not supported
+
+### Input Format Support
+
+| Format | vLLM | TRT-LLM | SGLang |
+|--------|------|---------|--------|
+| HTTP/HTTPS URL | ✅ | ✅ | ✅ |
+| Data URL (Base64) | ✅ | ❌ | ❌ |
+| Pre-computed Embeddings (.pt) | ❌ | ✅ | ❌ |
+
+## Architecture Patterns
+
+Dynamo supports several deployment patterns for multimodal inference based on two dimensions:
+
+1. **Encoding**: Is media encoding handled inline (within prefill) or by a separate **Encode Worker**?
+   - *Inline*: Simpler setup, encoding happens in the prefill worker
+   - *Separate (EPD)*: Dedicated encode worker transfers embeddings via **NIXL (RDMA)**, enabling independent scaling
+
+2. **Prefill/Decode**: Are prefill and decode in the same worker or separate?
+   - *Aggregated*: Single worker handles both prefill and decode
+   - *Disaggregated*: Separate workers for prefill and decode, with KV cache transfer between them
+
+These combine into four deployment patterns:
+
+### EPD - Simple Aggregated
+
+All processing happens within a single worker - the simplest setup.
+
+```text
+HTTP Frontend (Rust)
+    ↓
+Worker (Python)
+    ↓ image load + encode + prefill + decode
+Response
+```
+
+| Component | Purpose |
+|-----------|---------|
+| Frontend (Rust) | HTTP entry point, tokenization, image URL preprocessing |
+| Worker | Complete inference pipeline (encode + prefill + decode) |
+
+**When to use:** Quick setup, smaller models, development/testing.
+
+### E/PD - Encode Separate
+
+Encoding happens in a separate worker; prefill and decode share the same engine.
+
+```text
+HTTP Frontend (Rust)
+    ↓
+Processor (Python)
+    ↓ tokenizes, extracts media URL
+Encode Worker (Python)
+    ↓ downloads media, generates embeddings, NIXL transfer
+PD Worker (Python)
+    ↓ receives embeddings via NIXL, prefill + decode
+Response
+```
+
+| Component | Purpose |
+|-----------|---------|
+| Frontend (Rust) | HTTP entry point |
+| Processor (Python) | Tokenization, extracts media URLs |
+| Encode Worker | Media encoding, embeddings generation |
+| PD Worker | Prefill + Decode with embeddings |
+
+**When to use:** Offload vision encoding to separate GPU, scale encode workers independently.
+
+### E/P/D - Full Disaggregation
+
+Full disaggregation with separate workers for encoding, prefill, and decode.
+
+```text
+HTTP Frontend (Rust)
+    ↓
+Processor (Python)
+    ↓ tokenizes, extracts media URL
+Encode Worker (Python)
+    ↓ downloads media, generates embeddings, NIXL transfer
+Prefill Worker (Python)
+    ↓ receives embeddings via NIXL, prefill only, KV cache transfer
+Decode Worker (Python)
+    ↓ decode only, token generation
+Response
+```
```

Comment:
I think this workflow here is vllm E/P/D. For sglang it's slightly different, it will be `Frontend → Processor → Encode → DECODE Worker → Prefill Worker`